packages/use-presence/src/index.tsx

```diff
@@ -107,3 +109,57 @@ export default function usePresence(
     isExiting
   };
 }
+
+export type UseUniqueDataPresenceOptions<T> = {
+  /** Check if two instances of type T are equal, defaults to a === comparison */
+  equalityCheck?(a: T, b: T): boolean;
+  /** Check if an instance of type T is considered valid to render, defaults to Boolean function */
+  validationCheck?(a: T): boolean;
+} & UsePresenceOptions;
+
+function defaultEqualityCheck<T>(a: T, b: T) {
+  return a === b;
+}
+function defaultValidationCheck<T>(a: T) {
+  return Boolean(a);
+}
+
+export function useUniqueDataPresence<T>(
+  dataInput: T,
```

Comment:
Maybe we could use `item` or `value` as a name to refer to individual items? I personally also use expressive names for generics, I think that helps to understand the intent a bit better on the consumer side.

Reply:
I like `item`, updated 👍